util/conditions/patch.go

```diff
@@ -50,9 +50,14 @@ const (
 )
 
 // NewPatch returns the list of Patch required to align source conditions to after conditions.
+// Can return nil if either of the parameters -  before or after - is nil.
```

Comment:
This is problematic - a nil parameter just returns what looks like an empty patch to the user but:
1) It's no worse than before
2) We can't return an error here as it would be an API change. 

I think we can leave it like this for now and, if agreed, maybe add an error return to this function in a future release?

Reply:
I think we should introduce a new method and deprecate the current one.
What about NewPatchOperations or NewDiff (eventually after removal of the current method we can eventually re-use NewPatch again

Reply:
I reverted the changes in NewPatch and added NewPatchOperations which can return an error along with a Patch.
NewPatch is deprecated and replaced in its usage in our code here.